### ISSUE TITLE
Suggestion: say "field is never read" instead of "field is never used" 

### ISSUE THREAD

**Issue by xpe** [rust-lang/rust#64465]
As of rustc 1.37.0, this code:

```rs
pub struct Owl { tree: u16 }

fn exist(tree: u16) -> bool { tree > 0 && tree <= 3 }

impl Owl {
    pub fn find(tree: u16) -> Option<Owl> {
        if exist(tree) { Some(Owl { tree }) } else { None }
    }
}

fn main() { Owl::find(1); }
```

gives the warning:

```
warning: field is never used: `tree`
 --> src/main.rs:5:18
```

I understand that the compiler can optimize out the `tree` field. Forgive me for stating the obvious; namely, the warning doesn't mention anything about "optimizing away" the field; it only says "field is never used". A user would be justified in thinking: "this warning doesn't make sense; the `Owl::find` method most certainly **uses** the `tree` field.

**Suggestion**

I'd suggest changing the warning to say "warning: field is never read".

**Rationale**

By the common understanding of the word 'use', the `tree` field indeed is 'used', because the word 'use' can mean 'read' or 'write'. Being more specific (i.e. saying "field is never read") will help users understand what **kind** of use the compiler means.

This issue has been assigned to @CosineP via [this comment](https://github.com/rust-lang/rust/issues/64465#issuecomment-554629019).

**Comment by lzutao** [comment]
Ping @XiangQingW , do you still working on it?

**Comment by XiangQingW** [comment]
@lzutao No. I'm working on other issues currently. Please take it if you want.

**Comment by vlabo** [comment]
I would like to fix it. But it will be my first problem so it may take a while.

**Comment by CosineP** [comment]
i will take it? @rustbot claim? first time sorry

**Comment by jplatte** [comment]
@CosineP I think rostbot was confused by the questionmark after `claim` ;)
